src/backend_model_instance.cc

```diff
@@ -558,7 +559,7 @@ TritonModelInstance::PrepareRequestsOrRespond(
   // If any errors occurred, respond with error for each request.
   if (!status.IsOk()) {
     for (auto& r : requests) {
-      InferenceRequest::RespondIfError(r, status, true /* release_requests */);
+      InferenceRequest::RespondIfError(r, status, true /* release_requests */, FailureReason::BACKEND);
```

Comment:
Can you add a test to confirm whether or not this `RespondIfError` call and the `TRITONBACKEND_ModelInstanceReportStatistics(..., success=false, ...)` is double counting the same error?